rpcserver/socket_reads.cc

```diff
@@ -125,13 +127,26 @@ namespace rpcserver {
         delete[] buf;
         return !error && proto.ParseFromString(proto_str);
     }
-
-
+	
+	/**
+     * Attempt to parse given protocol with given length from provided socket.
+     * Return whether the parse was successful.
+     */
+    bool read_proto(tcp::socket& sock, ::google::protobuf::Message& proto, uint64_t len) {
+        asio::error_code error;
+        char* buf = new char[len];
```

Comment:
@nichhk Were you thinking of the fill-constructor for a string? (i.e. std::string (size_t n, char c);) There doesn't appear to be a simple constructor with only a length parameter.